### PR TITLE
🎉 New Source: Workable (low-code SDK)

### LINKED ISSUE #19: Source: Workable

### PR DESCRIPTION
**Pull request by deanmalan**
## What

Add support for Workable as a source to close https://github.com/airbytehq/connector-contest/issues/19.

## How

Using the low-code SDK.

## Recommended reading order
1. `airbyte-integrations/connectors/source-workable/source_workable/spec.yaml`
2. `airbyte-integrations/connectors/source-workable/source_workable/workable.yaml`

## 🚨 User Impact 🚨

None.

## Pre-merge Checklist

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

Don't have any.

</details>

<details><summary><strong>Integration</strong></summary>

Not sure how to run other integrations tests than the acceptance ones below?

</details>

<details><summary><strong>Acceptance</strong></summary>

Acceptance tests:
Screenshot 2022-10-15 at 22 12 26

</details>

### CODE REVIEW

**Comment by harshithmullapudi** [comment]
Thanks for the contribution. I will take a look at this tomorrow.

File: airbyte-integrations/connectors/source-workable/source_workable/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  additionalProperties: true
+  properties:
+    api_key:
+      type: string
+      description: Your Workable API Key. See <a href="https://workable.readme.io/reference/generate-an-access-token">here</a>.
+      airbyte_secret: true
+    account_subdomain:
+      type: string
+      description: Your Workable account subdomain, e.g. https://your_account_subdomain.workable.com.
+    created_after_date:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-workable/source_workable/spec.yaml:19]: Usually this parameter is called `start_date`

File: airbyte-integrations/connectors/source-workable/source_workable/workable.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      name: "stages"
+      primary_key: "slug"
+      path: "/spi/v3/stages"
+      field_pointer: ["stages"]
+  recruiters_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "recruiters"
+      primary_key: "id"
+      path: "/spi/v3/recruiters"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-workable/source_workable/workable.yaml:70]:
```suggestion
      path: "/spi/v3/recruiters"
```

Why `/spi/v3/` is not in the base url?

File: docs/integrations/sources/workable.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+* [Recruiters](https://workable.readme.io/reference/recruiters)
+
+## Data type map
+
+| Integration Type    | Airbyte Type |
+| :------------------ | :----------- |
+| `string`            | `string`     |
+| `integer`, `number` | `number`     |
+| `array`             | `array`      |
+| `object`            | `object`     |
```
Comment (marcosmarxm) [inline docs/integrations/sources/workable.md:57]: Remove this

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Looks great, I made a few comments to you improve the connector. @harshithmullapudi please make a deep review today.

File: airbyte-integrations/connectors/source-workable/source_workable/workable.yaml (hunk 2 above)
Comment (delenamalan) [reply airbyte-integrations/connectors/source-workable/source_workable/workable.yaml:70]: Pagination didn't work correctly with `/spi/v3/` in the base URL.

Comment (delenamalan) [reply airbyte-integrations/connectors/source-workable/source_workable/workable.yaml:70]: I've created an issue for it here now: https://github.com/airbytehq/airbyte/issues/18136.

**Comment by marcosmarxm** [comment]
@delenamalan is it possible to share the credentials to run tests?

File: airbyte-integrations/connectors/source-workable/integration_tests/sample_config.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "api_key": "<api_key>",
+  "account_subdomain": "<account_subdomain>"
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-workable/integration_tests/sample_config.json:3]: start_date is needed here right?

File: airbyte-integrations/connectors/source-workable/source_workable/workable.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  jobs_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "jobs"
+      primary_key: "id"
+      path: "/spi/v3/jobs"
+      field_pointer: ["jobs"]
+  candidates_stream:
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-workable/source_workable/workable.yaml:51]: could you add comments here to the respective stream documentation

**Review by harshithmullapudi — CHANGES_REQUESTED** [review]

**Comment by harshithmullapudi** [comment]
Hey, were you able to run connector and tests in local if so could you share the credentials so that I can run the tests.

**Comment by marcosmarxm** [comment]
@harshithmullapudi please create an issue to unblockers team create internal sandbox account too.

**Review by harshithmullapudi — APPROVED** [review]

**Review by marcosmarxm — APPROVED** [review]
Thanks @delenamalan

**Comment by harshithmullapudi** [comment]
/publish connector=connectors/source-workable run-tests=false

**Pull request merged**